jina/optimizers/__init__.py

```diff
@@ -1,22 +1,35 @@
 from pathlib import Path
-from ..jaml import JAML
 
-from .flow_runner import MultiFlowRunner
-from jina.helper import colored
-from jina.logging import default_logger as logger
+from typing import Optional
+from collections import defaultdict
+
+from ..helper import colored
+from ..jaml import JAML
+from ..logging import default_logger as logger
 from .parameters import load_optimization_parameters
 
 
+if False:
+    from .flow_runner import MultiFlowRunner
+
+
 class EvaluationCallback:
-    def __init__(self, eval_name=None):
+    """Callback for storing and calculating evaluation metric."""
+
+    def __init__(self, eval_name: Optional[str] = None):
+        """
+        :param eval_name: evaluation name as required by evaluator
+        """
         self.eval_name = eval_name
-        self.evaluation_values = {}
+        self.evaluation_values = defaultdict(float)
         self.n_docs = 0
 
     def get_fresh_callback(self):
```

Comment:
this should be a classmethod.

Reply:
For this to be a `staticmethod`, it needs to have the `args` needed to create it which we don't have in `OptunaOptmizer`

Reply:
u can extract it from the last `evaluationCallback`.

u would do `self.callback = EvaluationCallback.get_fresh_callback(self.callback.eval_name)`

Reply:
If we are using from the `self` it doesn't change Optimizer's behaviour since earlier it also uses from self. The intention was not to get a new callback with new op_name but the same op_name. Please confirm the change

Reply:
sure, no problem